### ISSUE TITLE
Problem with WebSocketClient Example

### ISSUE THREAD

**Issue by chessweb01** [Links2004/arduinoWebSockets#158]
I'm running the original WebSocketClient example on NodeMCU v1.0. 

Except for SSID, passwort for the WLAN router and hostname, port for the web server, nothing has been changed in the code.

This is what I see on the serial monitor:
[SETUP] BOOT WAIT 4...
[SETUP] BOOT WAIT 3...
[SETUP] BOOT WAIT 2...
[SETUP] BOOT WAIT 1...
f r0, scandone
f r0, scandone
...

The program appears to be stuck in the `while(WiFiMulti.run() != WL_CONNECTED)` loop and every second or so f r0, scandone is issued. 

The problem is not with my general NodeMCU and WiFi setup, as I was able to verify it with another program (https://blog.thesen.eu/wp-content/uploads/2015/04/ESP_Webserver_HTTP11.zip).

**Comment by Links2004** [comment]
is you network hidden? WiFiMulti only work on visibel networks.

you can change 
```cpp
while(WiFiMulti.run() != WL_CONNECTED)
```
to any other connection Methode that is working with you network.
the websocket lib has no dependency to WiFiMulti.

**Comment by chessweb01** [comment]
Thanks. Enabling SSID broadcast helped. But now I get [WSc] Disconnected! in the serial monitor every 100 ms or so. 

The problem is not the server. It is working fine when a websocket client runs in a browser.

**Comment by chessweb01** [comment]
In just stumbled upon issue #132 and the beginSocketIO method for socket.io servers. My server is written in golang using the gorilla websocket implementation (https://github.com/gorilla/websocket). 

What do I have to do to adapt the begin method to my server?

**Comment by Links2004** [comment]
on the first look gorilla look like a normal websocket server not a socket.io server.

I think:
```cpp
webSocket.begin("IP", 80, "/", "");
```
is the way to go.
if this not work please enable debug and post the output here.
1. Set debug port in IDE Menu
2. enable debug for Websockets (https://github.com/Links2004/arduinoWebSockets/blob/master/src/WebSockets.h#L30)

**Comment by chessweb01** [comment]
Didn't work with `webSocket.begin("IP", 80, "/", "");` Here is the debug output:

[WS-Client][sendHeader] sending header... Done (68328us).
[WS-Client][handleHeader] RX: HTTP/1.1 200 OK
[WS-Client][handleHeader] RX: Cache-Control: no-cache
[WS-Client][handleHeader] RX: Cache-Control: must-revalidate
[WS-Client][handleHeader] RX: Date: Sun, 08 Jan 2017 12:04:36 GMT
[WS-Client][handleHeader] RX: Content-Length: 714
[WS-Client][handleHeader] RX: Content-Type: text/html; charset=utf-8
[WS-Client][handleHeader] Header read fin.
[WS-Client][handleHeader] Client settings:
[WS-Client][handleHeader]  - cURL: /
[WS-Client][handleHeader]  - cKey: fjmxi3xMAuxWLKUzSND1IA==
[WS-Client][handleHeader] Server header:
[WS-Client][handleHeader]  - cCode: 200
[WS-Client][handleHeader]  - cIsUpgrade: 0
[WS-Client][handleHeader]  - cIsWebsocket: 0
[WS-Client][handleHeader]  - cAccept: 
[WS-Client][handleHeader]  - cProtocol: 
[WS-Client][handleHeader]  - cExtensions: 
[WS-Client][handleHeader]  - cVersion: 0
[WS-Client][handleHeader]  - cSessionId: 
[WS-Client][handleHeader] no Websocket connection close.
[WS-Client] client disconnected.
[WSc] Disconnected!
[WS-Client] connect ws...
[WS-Client] connected to kreidestaub.de:8080.
[WS-Client][sendHeader] sending header...
[WS-Client][sendHeader] handshake GET / HTTP/1.1
Connection: Upgrade
Upgrade: websocket
Sec-WebSocket-Version: 13
Sec-WebSocket-Key: GTJ6qBgEnqEbf+kPuwO31w==
Host: kreidestaub.de:8080
Origin: file://
User-Agent: arduino-WebSocket-Client

This output is repeated indefinitely. I hope it provides some clues as to what is going wrong.

**Comment by Links2004** [comment]
yes it does ;)
you are connection to a normal webserver, and getting a html page (the normal index page).
```
Content-Type: text/html; charset=utf-8
```

may you need to change the path (3th parameter) to match your server setup.
I think websockets are only handelt on a specific part / url.
I recommend to check your server config / the docs of gorilla for it.

**Comment by chessweb01** [comment]
Fixed it.

First off, my server requires /ws instead of / as path parameter. After that I got 403 permission denied instead of 200 OK from the server. The reason is the value of the origin field (file://). It must match the host (same origin policy). After I changed that in WebSocketsClient.cpp:449 the program works as far as the websocket connection is concerned.

Anyway, thanks for your help. Feel free to close this issue.
